### PR TITLE
Revert "Merge pull request #7943 from habitat-sh/cm/revert-tls"

### PR DESCRIPTION
**Pull request by christophermaier**
This is a second attempt at merging #7923.

All changes are by @davidMcneil; I'm just the messenger.

This reverts commit 54947732959ccec37da682fe4afa8752f1c25104, reversing
changes made to 59e3c3b4fdd3aed0fc915e28e2739f2e7bc536e7.

Signed-off-by: David McNeil <mcneil.david2@gmail.com>
Signed-off-by: Christopher Maier <cmaier@chef.io>

### CODE REVIEW

**Review by jeremymv2 — APPROVED** [review]

File: test/end-to-end/test_tls_ctl_gateway.ps1 [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            Invoke-NativeCommand hab sup run --ctl-server-certificate --ctl-server-key
+        } | Should -Throw
+    }
+
+    It "ctl gateway key and certificate generation" {
+        Invoke-NativeCommand hab sup secret generate-tls --subject-alternative-name localhost
+    }
+
+    It "ctl gateway TLS connection fails without TLS enabled client" {
+        Start-Supervisor -Timeout 45 -SupArgs @( `
```
Comment (smacfarlane) [inline test/end-to-end/test_tls_ctl_gateway.ps1:23]: Do these Start-Supervisor commands need to handle the $logfile parameter now?

**Review by smacfarlane — CHANGES_REQUESTED** [review]
One small change to the e2e test that I think is necessary :smile_cat:

File: test/end-to-end/test_tls_ctl_gateway.ps1 (hunk 1 above)
Comment (christophermaier) [reply test/end-to-end/test_tls_ctl_gateway.ps1:23]: Ah, right you are! Fix incoming.

**Review by smacfarlane — APPROVED** [review]

**Pull request merged**